task-1/src/Shape.ts

```diff
@@ -0,0 +1,37 @@
+import { Point } from './Point';
+
+export abstract class Shape {
+  protected color: string;
+  protected filled: boolean;
+  protected points: Point[];
+
+  constructor(points: Point[], color?: string, filled?: boolean);
+  constructor(points: Point[], color: string, filled: boolean) {
+    if (points.length < 3) {
+      throw new Error;
+    }
+
+    this.points = points;
+    this.color = color ?? "green";
+    this.filled = filled ?? true;
+  }
+
+  private isLastIndex = (index: number, arrayLength: number) => index === arrayLength - 1;
+
+  private getTotalPoints(): string {
+    return this.points.map((point: Point) => point.toString()).join(', ');
```

Comment:
You can replace it with just the `this.points.join(', ')`.
`toString` is called internally